src/Arcus.Observability.Correlation/IServiceCollectionExtensions.cs

```diff
@@ -11,29 +11,60 @@ namespace Microsoft.Extensions.DependencyInjection
     // ReSharper disable once InconsistentNaming
     public static class IServiceCollectionExtensions
     {
+        /// <summary>
+        /// Adds operation and transaction correlation to the application using the <see cref="DefaultCorrelationInfoAccessor"/>
+        /// </summary>
+        /// <param name="services">The services collection containing the dependency injection services.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
+        public static IServiceCollection AddCorrelation(this IServiceCollection services)
+        {
+            Guard.NotNull(services, nameof(services), "Requires a service collection to register the default correlation accessor to the application services");
+
+            return AddCorrelation<CorrelationInfo>(services);
+        }
+
         /// <summary>
         /// Adds operation and transaction correlation to the application using the <see cref="DefaultCorrelationInfoAccessor"/>
         /// </summary>
         /// <param name="services">The services collection containing the dependency injection services.</param>
         /// <param name="configureOptions">The function to configure additional options how the correlation works.</param>
+        [Obsolete("Use HTTP, messaging, or custom-specific correlation registration instead")]
         public static IServiceCollection AddCorrelation(
             this IServiceCollection services,
-            Action<CorrelationInfoOptions> configureOptions = null)
+            Action<CorrelationInfoOptions> configureOptions)
         {
             Guard.NotNull(services, nameof(services));
 
             return AddCorrelation(services, new DefaultCorrelationInfoAccessor(), configureOptions);
         }
 
+        /// <summary>
+        /// Adds operation and transaction correlation to the application using the <see cref="DefaultCorrelationInfoAccessor{TCorrelationInfo}"/>
+        /// </summary>
+        /// <typeparam name="TCorrelationInfo">The type of the <see cref="CorrelationInfo"/> model.</typeparam>
+        /// <param name="services">The services collection containing the dependency injection services.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
+        public static IServiceCollection AddCorrelation<TCorrelationInfo>(
```

Comment:
I don't immediately see how this must be used now ?
You need to create a type which inherits from CorrelationInfo, or do we have some predefined types ?

Reply:
Yes, this is used when you want to use a custom correlation model. This extension is used, for example, in the message pump where we use `MessageCorrelationInfo`.

Reply:
ok, thx